flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/ChunkSplitter.java

```diff
@@ -65,54 +66,76 @@ class ChunkSplitter {
     private final MySqlSourceConfig sourceConfig;
     private final MySqlSchema mySqlSchema;
 
+    private final JdbcConnection jdbcConnection;
+    private Table currentSplittingTable;
+    private Column splitColumn;
+    private RowType splitType;
+    private Object[] minMaxOfSplitColumn;
+    private long approximateRowCnt;
+
     public ChunkSplitter(MySqlSchema mySqlSchema, MySqlSourceConfig sourceConfig) {
         this.mySqlSchema = mySqlSchema;
         this.sourceConfig = sourceConfig;
+        this.jdbcConnection = openJdbcConnection(sourceConfig);
     }
 
-    /** Generates all snapshot splits (chunks) for the give table path. */
-    public Collection<MySqlSnapshotSplit> generateSplits(TableId tableId) {
-        try (JdbcConnection jdbc = openJdbcConnection(sourceConfig)) {
-
-            LOG.info("Start splitting table {} into chunks...", tableId);
-            long start = System.currentTimeMillis();
-
-            Table table = mySqlSchema.getTableSchema(jdbc, tableId).getTable();
-            Column chunkKeyColumn =
-                    ChunkUtils.getChunkKeyColumn(table, sourceConfig.getChunkKeyColumn());
-            final List<ChunkRange> chunks;
-            try {
-                chunks = splitTableIntoChunks(jdbc, tableId, chunkKeyColumn);
-            } catch (SQLException e) {
-                throw new FlinkRuntimeException("Failed to split chunks for table " + tableId, e);
-            }
+    /** This method should be invoked before getting chunks for a table. */
+    public void switchSplittingTable(TableId tableId) throws SQLException {
```

Comment:
We can let `ChunkSplitter` be stateful,  the `ChunkSplitter` can restore from `ChunkSplitterState` and will produce a `ChunkSplitterState` when call `snapshotState()`, the `ChunkSplitterState` consists of multiple fields.  In this way, we can make the code more clean.